chia/types/blockchain_format/vdf.py

```diff
@@ -96,3 +100,64 @@ class CompressibleVDFField(IntEnum):
     ICC_EOS_VDF = 2
     CC_SP_VDF = 3
     CC_IP_VDF = 4
+
+
+def compress_future(
+    disc: str,
+    output: bytes,
+    input: bytes,
+    proof: bytes,
+    number_of_iterations: uint64,
+    proof_type: int,
+) -> B:
+    return get_b_from_n_wesolowski(
+        disc,
+        input,
+        output + proof,
+        number_of_iterations,
+        proof_type,
+    )
+
+
+def compress_output(
+    disc_size: int,
+    challenge: bytes32,
+    input: ClassgroupElement,
+    output: ClassgroupElement,
+    proof: VDFProof,
+    number_of_iterations: uint64,
+    executor: ProcessPoolExecutor,
+) -> Future[B]:
+
+    future = executor.submit(
+        compress_future,
+        str(get_discriminant(challenge, disc_size)),
+        bytes(output.data),
+        bytes(input.data),
+        bytes(proof.witness),
+        number_of_iterations,
+        proof.witness_type,
+    )
+
+    return future
+
+
+def verify_compressed_vdf(
+    constants: ConsensusConstants,
+    challenge: bytes32,
+    vdf_input: ClassgroupElement,
+    vdf_output: B,
+    proof: VDFProof,
+    number_of_iterations: uint64,
+) -> Tuple[bool, ClassgroupElement]:
+    if proof.witness_type + 1 > constants.MAX_VDF_WITNESS_SIZE:
+        raise Exception(f"invalid witness type")
+    x, y = verify_n_wesolowski_with_b(
```

Comment:
what is x, is that a bool for whether the proof was valid? If so, we should rename it. 
We can also use type annotations in the line above, to make it more clear:
```
is_valid: bool
output_y: ClassgroupElement
```

Reply:
What is returned if the proof is false?
Do we have any tests for this function?